pkg/kube/controllers/extendedjob/job_creator.go

```diff
@@ -89,23 +91,48 @@ func (j jobCreatorImpl) Create(ctx context.Context, eJob ejv1.ExtendedJob, names
 		},
 	}
 
-	// Set serviceaccount to the pod
-	template.Spec.ServiceAccountName = serviceAccountName
-
-	err = j.client.Create(ctx, serviceAccount)
-	if err != nil {
+	if err := j.client.Create(ctx, serviceAccount); err != nil {
 		if !apierrors.IsAlreadyExists(err) {
 			return false, errors.Wrapf(err, "could not create service account for pod in ejob %s.", eJob.Name)
 		}
 	}
 
-	err = j.client.Create(ctx, roleBinding)
-	if err != nil {
+	if err := j.client.Create(ctx, roleBinding); err != nil {
```

Comment:
What happens if sa creation does not finish before this rb creation?

Reply:
Nothing. The `RoleBinding` doesn't require the `ServiceAccount` or the `Role` to exist.

```sh
$ kubectl create namespace foo
namespace/foo created

$ kubectl apply -f - <<EOF
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: read-pods
  namespace: foo
subjects:
- kind: ServiceAccount
  name: does-not-exist
  namespace: foo
roleRef:
  kind: Role
  name: does-not-exist
  apiGroup: rbac.authorization.k8s.io
EOF
rolebinding.rbac.authorization.k8s.io/read-pods created
```